src/main/scala/nl.knaw.dans.easy.ccw/EasyCollectCurationWorkApp.scala

```diff
@@ -15,17 +15,70 @@
  */
 package nl.knaw.dans.easy.ccw
 
-import scala.util.{ Success, Try }
+import better.files.File
+import nl.knaw.dans.lib.logging.DebugEnhancedLogging
+import org.apache.commons.configuration.PropertiesConfiguration
 
-class EasyCollectCurationWorkApp(configuration: Configuration)  {
+import scala.collection.JavaConverters._
+import scala.language.postfixOps
+import scala.util.Try
 
-  def run(): Try[Unit] = {
+class EasyCollectCurationWorkApp(commonCurationArea: File, managerCurationDirString: String, datamanagerProperties: PropertiesConfiguration) extends DebugEnhancedLogging {
 
+  private def isCurated(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("curation.performed", "") == "yes"
+  }
 
+  private def noDescription(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("state.description", "").isEmpty
+  }
 
+  private def noDatamanagerUserID(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("curation.datamanager.userId", "").isEmpty
+  }
 
-    ???
+  private def noDatamanagerEmail(depositProperties: PropertiesConfiguration): Boolean = {
+    depositProperties.getString("curation.datamanager.email", "").isEmpty
   }
 
+  private def collectDeposit(datamanager: DatamanagerId, deposit: File): Unit = {
+    val depositProperties = new PropertiesConfiguration(deposit / "deposit.properties" toJava)
+    val state = depositProperties.getString("state.label", "")
```

Comment:
if no state is given, `""` should not be the correct default, I think.
Better wrap it in an `Option` and handle it as an error in the decision tree below.

Reply:
"" is the correct default, because here is checked that the property is not empty and that it exists.  Now the result is false also when the property doesn't exist, which is correct.
Option does not work here, because Option("").isEmpty is false.